pkg/interfaces/controllers/api_controller.go

```diff
@@ -59,3 +91,39 @@ func (r *APIReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&kctlrv1beta1.API{}).
 		Complete(r)
 }
+
+func (r *APIReconciler) getApiListStatus(ctx context.Context, req ctrl.Request) (*v1.ConfigMap, error) {
+	apiListStatus := &v1.ConfigMap{}
+	namespacedName := k8stypes.NamespacedName{Namespace: req.Namespace, Name: apiListStatusConfigMap}
+	if err := r.Client.Get(ctx, namespacedName, apiListStatus); err != nil && errors.IsNotFound(err) {
+		if apiListStatus, err = r.createApiListStatus(ctx, namespacedName); err != nil {
+			return nil, err
+		}
+	}
+	return apiListStatus, nil
+}
+
+func (r *APIReconciler) patchApiListStatus(ctx context.Context, apiListStatus *v1.ConfigMap, newValues map[string]string) error {
+	apiListStatusPatch := client.MergeFrom(apiListStatus.DeepCopy())
```

Comment:
Any specific reason to do the merging manually?

It should be enough to update the existing CM: you need to read it first, then update the data and then `r.Client.Update(ctx, configmap)`

Reply:
No particular reason besides following the `Patch` example from https://sdk.operatorframework.io/docs/building-operators/golang/references/client/#patch. Maybe it's better to use `Update` instead of `Patch`?

Reply:
Both ways would do the same for this particular case. 

Using the Update method, you will not need to call `apiListStatusPatch := client.MergeFrom(apiListStatus.DeepCopy())`

For me, patching is low level operation that can be used when you want to specify the "update" behavior at the highest level. For me Update is semantically like SET operation. You tell the cluster the desired object. The client or the cluster makes it happen. On the other had, patching is like telling the delta to be applied. The "MergeFrom" is a helper function to compute the delta for you and still let's you add patching options.

Reply:
Makes total sense, I'll use `Update`